hibike/spawn_virtual_devices.py

```diff
@@ -0,0 +1,38 @@
+import csv
+import subprocess
+import shlex
+import re
+import os
+import atexit
+import time
+
+popens = []
+
+def get_virtual_ports():
+    socat = subprocess.Popen(shlex.split("socat -d -d pty,raw,echo=0 pty,raw,echo=0"), stderr=subprocess.PIPE)
+    popens.append(socat)
+    lines = [str(socat.stderr.readline()), str(socat.stderr.readline())]
```

Comment:
(copied from previous PR)
`str` is not the right function for converting bytestrings to strings. Try `str(b"test")` in the REPL, it won't do the right thing.

If fixing bytestrings is indeed needed, you need to use `encode` to specify ascii, utf-8, or another encoding.

Given that `str` cannot possibly cast correctly, then either:
1. You can remove the cast
2. You should use the correct byte-to-str conversion